### PR TITLE
Recent Authors List fix

### LINKED ISSUE #389: Repetition in recent authors sidebar
image

### PR DESCRIPTION
**Pull request by govindjeevan**
Fixes #389 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkboxe write 'x' within the square brackets)

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/models/map.rb [hunk 1]
```diff
@@ -87,8 +87,8 @@ def latest_export
   def self.authors(limit = 50)
     Map.limit(limit)
        .order("maps.id DESC")
-       .where('password = "" AND archived = "false"')
-       .collect(&:author)
+       .where('password = "" AND archived = "false" AND author!="anonymous"')
+       .collect(&:author).uniq
```
Comment (jywarren) [inline app/models/map.rb]: this should be good, although i think in ActiveRecord we could also do this, maybe?

```suggestion
       .distinct.collect(&:author)
```

Comment (jywarren) [reply app/models/map.rb]: oh wait. that's not right. Um, instead -- maybe this? 
```suggestion
       .group(:author).collect(&:author)
```

Comment (govindjeevan) [reply app/models/map.rb]: How about
.group(:author).pluck(:author)
?

**Comment by sashadev-sky** [comment]
@govindjeevan sorry to get back to you on this late!

@jywarren I don't think any of the query suggestions will work probably because it interferes with some javascript code somewhere but @govindjeevan's original implementation worked so this is mergable if you want to go with it! 

It solves two issues currently appearing in recent authors: showing anonymous authors and author names appear on the list multiple times. So I think it would be a good start to merge this

File: app/models/map.rb (hunk 1 above)
Comment (jywarren) [reply app/models/map.rb]: Yes, nice! Let's do that! Sorry i missed your comment originally!

Comment (jywarren) [reply app/models/map.rb]:
```suggestion
       .group(:author).pluck(:author)
```

Comment (jywarren) [reply app/models/map.rb]: Try that!

**Comment by cesswairimu** [comment]
I am sorry, we redesigned the front page and the page that previously displayed 'recent authors' is no longer being rendered on the front page. Thanks for working on this and you are welcome to checkout other issues. Thanks .Closing this

**Pull request closed without merging**